.github/workflows/deploy.yml

```diff
@@ -0,0 +1,54 @@
+name: deploy
+
+on:
+  # Run on pushes to the default branch.
+  push:
+    branches:
+      - main
+  schedule:
+    - cron: "0 0 * * 0"
+
+# Declare default permissions as read only.
+permissions: read-all
+
+env:
+  # Keep for Dart SDK reporting
+  PUB_ENVIRONMENT: bot.github
+  # LTS
+  NODE_VERSION: '20'
+  # Tool location
+  BASE_DIR: ${{ github.workspace }}
+  TOOL_DIR: ${{ github.workspace }}/tool
+
+jobs:
+  deploy:
+    name: Deploy production site to Firebase hosting
+    permissions:
+      checks: write
+      pull-requests: write
+    if: ${{ github.event_name == 'push' &&
+      github.ref == 'refs/heads/main' &&
+      github.repository == 'dart-lang/site-www' }}
+    runs-on: ubuntu-latest
+    env:
+      FIREBASE_TOKEN: ${{ secrets.FIREBASE_TOKEN }}
+      FIREBASE_PROJECT: default
+    steps:
+      - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11
+        with:
+          submodules: recursive
+      - run: make build
+      - run: make write-prod-robots
+      - uses: actions/setup-node@b39b52d1213e96004bfcb1c61a8a6fa8ab84f3e8
+        with:
+          node-version: ${{ env.NODE_VERSION }}
+      - run: npm install -g firebase-tools@13.0.2
+      - uses: dart-lang/setup-dart@b64355ae6ca0b5d484f0106a033dd1388965d06d
+        with:
+          sdk: stable
+      - uses: FirebaseExtended/action-hosting-deploy@120e124148ab7016bec2374e5050f15051255ba2
```

Comment:
nit: Feel free to add a TODO for either @khanhnwin or myself to migrate this to cloud build, similar to the TODO in flutter/website!